Validate document count

steventux opened the pull request:
https://www.pivotaltracker.com/story/show/48238727
Adds validation to the document count for document based transactions.

Commit:
Validate document count

It's currently possible to legalise a document by post where the document count is zero.
This adds validation to catch any calculations where the document count is present and less than 1.

Commit:
Test zero document count transaction

Commit:
Test that a document count of zero throws the approriate error

alext commented on lib/transaction_calculator.rb:
```diff
@@ -7,8 +7,8 @@ def initialize(transaction)
-    if document_count
-      raise Transaction::InvalidDocumentCount unless values[:document_count].to_s =~ %r{\A[0-9]*\z}
+    if values.has_key?(:document_count)
+      raise Transaction::InvalidDocumentCount unless document_count > 0
```
Do we not want to also raise an error if no document_count is provided?

Commit:
Add invalid registration count exception

Commit:
Remove document_count check

The document count is currently always present in transactions,
there's no need to test for this key.

Commit:
Validate registration count where approriate
